submitter/sqs.py

```diff
@@ -1,76 +1,72 @@
+import json
+import logging
+
 import boto3
-import click
+from dspace.client import DSpaceClient
 
+from submitter import config
+from submitter.submission import Submission
 
-def message_loop(input_queue, output_queue, wait):
-    msgs = retrieve(input_queue, wait)
+logger = logging.getLogger(__name__)
 
-    if len(msgs) > 0:
-        click.echo(len(msgs))
-        process(msgs, output_queue)
-        message_loop(input_queue, output_queue, wait)
-    else:
-        click.echo("No messages received")
 
+def message_loop(queue, wait):
+    logger.info("Message loop started")
+    msgs = retrieve(queue, wait)
 
-def process(msgs, output_queue):
-    for message in msgs:
-        click.echo(message.message_attributes)
-        click.echo(message.body)
-        print("Do all the dspace submission stuff here")
+    if len(msgs) > 0:
+        process(msgs)
+        message_loop(queue, wait)
+    else:
+        logger.info("No messages available in queue %s", queue)
 
-        # faking it with always succeeding for now... creating of this status
-        # dict is likely better moved to our upcoming submission class but this
-        # was convenient for initial testing
-        status = {
-            "PackageSource": {
-                "DataType": "String",
-                "StringValue": message.message_attributes.get("PackageSource").get(
-                    "StringValue"
-                ),
-            },
-            "PackageID": {
-                "DataType": "String",
-                "StringValue": message.message_attributes.get("PackageID").get(
-                    "StringValue"
-                ),
-            },
-            "status": {"DataType": "String", "StringValue": "success"},
-            "handle": {
-                "DataType": "String",
-                "StringValue": "https://example.com/handle/this",
-            },
-        }
 
-        # write result to output
-        write(status, output_queue)
+def process(msgs):
+    client = DSpaceClient(config.DSPACE_API_URL)
+    client.login(config.DSPACE_USER, config.DSPACE_PASSWORD)
```

Comment:
I did think about doing that...it means passing the client around a lot because it actually gets called at a fairly low-level point in the whole message processing process. Not necessarily a bad thing but maybe not ideal either? I'd like to chat about this more at some point.